openwisp_controller/config/models.py

```diff
@@ -107,6 +108,23 @@ def get_temp_config_instance(self, **options):
         c.device = self
         return c
 
+    def save(self, *args, **kwargs):
```

Comment:
is there a specific reason for which you're defining this change here in openwisp-controller?

Consistent registration is implemented in django-netjsonconfig so you should move this logic in `django_netjsonconfig.base.models.Device`.

But you'll need to create a method that gets the shared secret: in django-netjsonconfig the method will return the global setting, in openwisp-controller it will return the `org_settings.shared_secret` like you did here.